Purchases/Networking/RCHTTPClient.m

```diff
@@ -10,13 +10,16 @@
 #import "RCLogUtils.h"
 #import "RCHTTPStatusCodes.h"
 #import "RCSystemInfo.h"
+#import "RCHTTPRequest.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 @interface RCHTTPClient ()
 
 @property (nonatomic) NSURLSession *session;
-@property (nonatomic, weak) RCSystemInfo *systemInfo;
+@property (nonatomic) RCSystemInfo *systemInfo;
+@property (nonatomic) NSMutableArray<RCHTTPRequest *> *queuedRequests;
+@property (nonatomic, nullable) RCHTTPRequest *currentSerialRequest;
```

Comment:
what I liked about handling this at a very low level is that you don't have to worry about chaining stuff together in the completion blocks - you just make requests one after another and the httpClient makes sure that they're not done concurrently

Reply:
this way we also set the requests to serial or concurrent directly per endpoint, in `RCBackend`, which felt like a reasonable way to tie the info - it's likely that if you need one request to be done serially, any requests to the same endpoint would also need to be done serially.